### PR TITLE
ci: only run builds and tests if rust files were modified

### PR DESCRIPTION
**Pull request by q9f**
Assumes tests and builds pass if no toml, lock, and rs file is modified.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -29,21 +29,31 @@ jobs:
...
+          fetch-depth: 0
       - name: Install Toolchain
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
           target: wasm32-unknown-unknown
           override: true
+      - name: Check for Rust changes
+        run: |
+          echo "::set-output name=files::$(git --no-pager diff --name-only origin/main...${{ github.event.pull_request.head.sha }} | egrep '(.toml)|(.lock)|(.rs)$' | wc -l | tr -d ' ')"
```
[the remark this answers is not in the document]
Comment (q9f) [reply .github/workflows/rust.yml:44]: - `rust-toolchain`: should never really change, should it?
- `.github/workflows`: I don't see how this affects the underlying Rust project. (The entire point of this PR is to work on docs, ci, etc. without having to wait 30 minutes for something that will succeed anyways.)
- `scripts`: same as CI?

I agree on the `Makefile`, though.

[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply .github/workflows/rust.yml:44]: * `rust-toolchain` theoretically it shouldn't, in practice we had it for some time on nightly when dealing with `ref-fvm`. 
* `.github/workflows` say you change Tarpaulin flags and you make a typo. Your PR will be green and once merged, anyone branching out of the `main` will have red builds until the issue is fixed (though it should've been caught in the original PR). Am I missing something?
* `scripts` - same, the command for linting calls `make lint` which will run `scripts/license.sh`. The errors made in this file won't be caught until a PR with Rust code changes.

[the remark this answers is not in the document]
Comment (q9f) [reply .github/workflows/rust.yml:44]: fair enough

[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply .github/workflows/rust.yml:44]: @q9f don't get me wrong, the idea is really good and would definitely facilitate docs development. Maybe we could go the other way - for example, if a PR changes **only** markdown files then the PR workflows shouldn't get triggered? What do you think?

[the remark this answers is not in the document]
Comment (q9f) [reply .github/workflows/rust.yml:44]: I think for now it's cleaner to just run all pipelines all the time.

**Pull request closed without merging**